module/documents/advancement/size.mjs

```diff
@@ -0,0 +1,82 @@
+import Advancement from "./advancement.mjs";
+import SizeConfig from "../../applications/advancement/size-config.mjs";
+import SizeFlow from "../../applications/advancement/size-flow.mjs";
+import { SizeConfigurationData, SizeValueData } from "../../data/advancement/size.mjs";
+
+/**
+ * Advancement that handles player size.
+ */
+export default class SizeAdvancement extends Advancement {
+
+  /** @inheritdoc */
+  static get metadata() {
+    return foundry.utils.mergeObject(super.metadata, {
+      dataModels: {
+        configuration: SizeConfigurationData,
+        value: SizeValueData
+      },
+      order: 5,
+      icon: "systems/dnd5e/icons/svg/size.svg",
+      title: game.i18n.localize("DND5E.AdvancementSizeTitle"),
+      hint: game.i18n.localize("DND5E.AdvancementSizeHint"),
+      validItemTypes: new Set(["race"]),
+      apps: {
+        config: SizeConfig,
+        flow: SizeFlow
+      }
+    });
+  }
+
+  /* -------------------------------------------- */
+  /*  Instance Properties                         */
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  get levels() {
+    return [0];
```

Comment:
I thought this wouldn't apply to advancements granted by non-class and non-subclass items?

Reply:
This is still relevant, it just refers to character level rather than class level. `0` here indicates that this should be applied regardless of current character level.